Classes/Command/CacheCommandController.php

```diff
@@ -122,4 +129,25 @@ public function listGroupsCommand()
                 break;
         }
     }
+
+    /**
+     * Flush cache for $cacheType with data handler implementation
+     *
+     * @param string $cacheType either "system" or "all"
+     * @return void
+     * @throws \UnexpectedValueException
+     */
+    public function flushCacheWithDataHandlerCommand($cacheType)
+    {
+        if (!in_array($cacheType, ['system', 'all'], true)) {
+            throw new \UnexpectedValueException('Only cache type "system" or "all" allowed. Given "'. $cacheType . '"');
+        }
+
+        /** @var \TYPO3\CMS\Core\DataHandling\DataHandler $dataHandler */
+        $dataHandler = $this->objectManager->get('TYPO3\\CMS\\Core\\DataHandling\\DataHandler');
+        $dataHandler->start([], []);
+        $dataHandler->clear_cacheCmd($cacheType);
```

Comment:
necessary code for commands should go into a service class. Since we have a service class for these commands already, we should add required things there